src/Codeception/Lib/InnerBrowser.php

```diff
@@ -1063,11 +1063,19 @@ protected function getFormFor(Crawler $node)
      */
     protected function getFormValuesFor(Form $form)
     {
+        $formNodeCrawler = new Crawler($form->getFormNode());
         $values = [];
         $fields = $form->all();
         foreach ($fields as $field) {
-            if ($field instanceof FileFormField || $field->isDisabled() || !$field->hasValue()) {
-                continue;
+            if ($field instanceof FileFormField || $field->isDisabled()
+                ||
+                !$field->hasValue()
+            ) {
+                // if unchecked a checkboc and if there is hidden input with same name to submit unchecked value
+                $hiddenInput = $formNodeCrawler->filter('input[type=hidden][name="'.$field->getName().'"]');
```

Comment:
This will also return inputs that are `disabled`, meaning we need to add a check for that.

This makes it troubling; it essentially means that all of the fall through logic that should be (and partially is) in DOM Crawler needs to be reimplemented here...

Reply:
For `disabled` main inputs (here `$field` var), `continue;`  statement will be executed . Also see conditions in above `if` statement. Agree with you that `filter()` here will grab also disabled input but in real world app it is very less likely to have input that is hidden+disabled.

Reply:
Sure; but that's a trade off that needs to be clear. Also isn't it simple to alter the call to `filter()` to skip disabled inputs?

Reply:
Yes there is way to filter out disabled input. Did that in https://github.com/Codeception/lib-innerbrowser/pull/36/commits/39f3e82f014e6e00b0ad7b602d117f96a2ffe391